src/app/components/main/Main.types.ts

```diff
@@ -1,4 +1,8 @@
+import { GetProductsUrlParams } from 'api/product/product.types';
+
 export type MainProps = {
   isScreenMobile: boolean;
   onFilterClick: () => void;
+  productsQuery: GetProductsUrlParams;
+  changeProductsQuery: (page: number) => void;
```

Comment:
> Main nie wie, że chodzi o zmianę strony, tylko o zmianę zapytania. Powinien dostać changeProductsQuery, bo wyżej dostaje productsQuery

Blisko. Ale pytanie, czy Main powinien dostać productsQuery? Jego nie interesuje skąd te dane o produkcie się wzięły. Więc właściwie one powinny się nazywać `productsParams`. Przypominajka - nie tworzymy zbędnych zależności pomiędzy komponentami ;)
A z perspektywy Maina, ten callback to jest ten sam callback, który jest wrzucany do paginacji. Czyli na zewnątrz, Main mówi - "Ziomki, tutaj mam taką funkcję co ją wykonam, kiedy (`on`) ktoś na paginację (`Page`) kliknie (`Click`).
Generalnie zasada jest taka - WSZYSTKIE callbacki zaczynają się od słowa `on`. Inaczej to jest funkcja sterująca komponentem, a do tego jeszcze nie doszliśmy (tutaj zajawka: https://pl.reactjs.org/docs/hooks-reference.html#useimperativehandle)

Reply:
Ok czaję